proto/provenance/name/v1/name.proto

```diff
@@ -47,6 +47,20 @@ message CreateRootNameProposal {
   bool   restricted  = 5;
 }
 
+// ModifyNameProposal details a proposal to update the address
+// and/or the restricted flag of a name.
+message ModifyNameProposal {
+  option (gogoproto.equal)            = false;
+  option (gogoproto.goproto_getters)  = false;
+  option (gogoproto.goproto_stringer) = false;
+
+  string title       = 1;
+  string description = 2;
+  string name        = 3;
+  string owner       = 4;
+  bool   restricted  = 5;
```

Comment:
Could use some field comments here to indicate what's the supposed to be old/existing, and which fields are new to be applied.

Reply:
Agreed